blocks/api/raw-handling/utils.js

```diff
@@ -204,13 +204,17 @@ function cleanNodeList( nodeList, doc, schema, inline ) {
 
 					// Strip invalid classes.
 					if ( node.classList.length ) {
-						const newClasses = classes.filter( ( name ) =>
-							node.classList.contains( name )
-						);
+						Array.from( node.classList ).forEach( ( name ) => {
+							if ( find( classes, ( item ) => {
+								return typeof item === 'string' ? item === name : item.test( name );
```

Comment:
I had trouble interpreting this `return` as not being part of the `forEach` but rather an inline predicate for the `find`. Can we extract this out? Or maybe normalize the `classes` schema item to a function?

```js
classes = classes.map( ( item ) => {
	if ( typeof item === 'string' ) {
		return ( className ) => className === item;
	} else if ( item instanceof RegExp ) {
		return ( className ) => item.test( className );
	}
} );

// ...

const newArray.from( node.classList ).forEach( ( name ) => {
	if ( ! classes.some( ( isMatch ) => isMatch( name ) ) {
		node.classList.remove( name );
	}
} );
```